flocker/node/_loop.py

```diff
@@ -429,6 +437,17 @@ def got_local_state(local_state):
             action = self.deployer.calculate_changes(
                 self.configuration, self.cluster_state, local_state
             )
+            if action == NoOp():
```

Comment:
I think it might be good to make it more explicit in the interface that if you *don't* compute some non-NoOp action then the system is going to sleep for a while.  ie, you may not want an implementation that passively waits for some condition to be discovered and then acts because the delay between iterations in this case may be much greater than you'd like. :question: